inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -19,78 +18,254 @@
-        # Creating the table
-        cur = conn.cursor()
-        # Check if the table already exists, if not, create the table
-        logger.debug("Checking if the table exist...")
-        cur.execute("SELECT name FROM sqlite_master WHERE type='table' AND name='repeated_task'")
-        if cur.fetchone() is None:
-            logger.debug("Table not exist. Creating the table.")
-            sql = ''' CREATE TABLE repeated_task (
-                                    request_id text,
-                                    cron_duration text,
-                                    cron_minutes text,
-                                    cron_hours text,
-                                    cron_day_month text,
-                                    cron_month text,
-                                    cron_day_week text,
-                                    manifests text
-                                ); '''
-            conn.execute(sql)  # Execute the SQL command
+    def _create_repeated_schedule(self, rs: RepeatedSchedule) -> None:
+        # Add the schedule to the single_schedule table
+        sql = ''' INSERT INTO repeated_schedule(request_id, cron_duration, cron_minutes, cron_hours, cron_day_month, cron_month, cron_day_week)
+                                VALUES(?,?,?,?,?,?,?); '''
+        try:
+            self._cursor.execute(sql, (rs.request_id, 
+                                        rs.cron_duration, 
+                                        rs.cron_minutes, 
+                                        rs.cron_hours, 
+                                        rs.cron_day_month, 
+                                        rs.cron_month, 
+                                        rs.cron_day_week))
+        except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+            raise DispatcherException(f"Error adding repeated schedule to database: {e}")
+        
+        schedule_id = self._cursor.lastrowid
+        if not schedule_id:
+            raise DispatcherException("No schedule id was added to the repeated_schedule table.")
+        logger.debug(f"Added repeated schedule with id: {str(schedule_id)}, request_id:{rs.request_id}, cron_duration: {rs.cron_duration}, cron_minutes: {rs.cron_minutes}, cron_hours: {rs.cron_hours}, cron_day_month: {rs.cron_day_month}, cron_month: {rs.cron_month}, cron_day_week: {rs.cron_day_week}") # noqa
+        
+        # Add the manifests to the manifest table
+        manifest_ids = self._insert_manifest_to_table(rs.manifests)
+           
+        # Add the schedule_id and manifest_id to the repeated_schedule_manifest table
+        self._insert_repeated_schedule_manifest_tables(schedule_id, manifest_ids)
+        self._conn.commit()
+        
+    def _insert_manifest_to_table(self, manifests: list[str]) -> list[int]:
+        # Add the manifest to the manifest table
+        manifest_ids: list[int] = []
+        
+        if len(manifests) == 0:
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Refactoring$</h3>

Consider handling the case where `manifests` is an empty list at the beginning of the `create_schedule` method to avoid deep nesting and early exit if validation fails.